### PR TITLE
📖 Improve ROSA quickstart with MachinePool and troubleshooting guide

### PR DESCRIPTION
**Pull request by serngawy**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

Enable the machinePool feature in order to be able to use ROSA machine pool feature

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #
https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4774

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [x] includes documentation
- [ ] includes [emojis](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
None
```

### CODE REVIEW

File: docs/book/src/topics/rosa/enabling.md [hunk 1]
```diff
@@ -4,5 +4,33 @@ To enable support for ROSA clusters, the ROSA feature flag must be set to true.
...
+export EXP_MACHINE_POOL="true"
 clusterctl init --infrastructure aws
+```
+
+## TroubleShoot
+To check the feature-gates for the capi contoller Run the following command.
+```shell
+$ kubectl get deploy capi-controller-manager -n capi-system -o yaml
+```
+check the feature gate container args it should has the MachinePool=true as below.
```
Comment (nrb) [inline docs/book/src/topics/rosa/enabling.md:16]:
```suggestion
the feature gate container arg it should have `MachinePool=true` as shown below.
```

File: docs/book/src/topics/rosa/enabling.md [hunk 2]
```diff
@@ -4,5 +4,33 @@ To enable support for ROSA clusters, the ROSA feature flag must be set to true.
...
+      containers:
+      - args:
+        - --feature-gates=MachinePool=true,ClusterResourceSet=true,ClusterTopology=true,RuntimeSDK=false,MachineSetPreflightChecks=false
+```
+
+To check the feature-gates for the capa contoller Run the following command.
+```shell
+$ kubectl get deploy capa-controller-manager -n capa-system -o yaml
+```
+check the feature gate container args it should has the ROSA=true as below.
```
Comment (nrb) [inline docs/book/src/topics/rosa/enabling.md:29]:
```suggestion
the feature gate arg should have `ROSA=true` as shown below.
```

**Comment by nrb** [comment]
Just a couple suggestions for wording, otherwise this looks good.

**Comment by nrb** [comment]
Could you update the commit message and title to reflect what the change does, rather than the issue number?

You can see what emoji to use for the PR title [here](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)

File: docs/book/src/topics/rosa/enabling.md [hunk 3]
```diff
@@ -4,5 +4,33 @@ To enable support for ROSA clusters, the ROSA feature flag must be set to true.
 
 ```shell
 export EXP_ROSA="true"
+export EXP_MACHINE_POOL="true"
 clusterctl init --infrastructure aws
+```
+
+## TroubleShoot
```
Comment (vincepri) [inline docs/book/src/topics/rosa/enabling.md:11]:
```suggestion
## Troubleshooting
```

File: docs/book/src/topics/rosa/enabling.md [hunk 4]
```diff
@@ -4,5 +4,33 @@ To enable support for ROSA clusters, the ROSA feature flag must be set to true.
...
+```shell
+$ kubectl get deploy capa-controller-manager -n capa-system -o yaml
+```
+check the feature gate container args it should has the ROSA=true as below.
+
+```yaml
+spec:
+    containers:
+      - args:
+        - --feature-gates=ROSA=true,...
```
Comment (vincepri) [inline docs/book/src/topics/rosa/enabling.md:35]: change to 2 spacing for better readability?

File: docs/book/src/topics/rosa/enabling.md [hunk 5]
```diff
@@ -4,5 +4,33 @@ To enable support for ROSA clusters, the ROSA feature flag must be set to true.
 
 ```shell
 export EXP_ROSA="true"
+export EXP_MACHINE_POOL="true"
 clusterctl init --infrastructure aws
+```
+
+## TroubleShoot
+To check the feature-gates for the capi contoller Run the following command.
```
Comment (vincepri) [inline docs/book/src/topics/rosa/enabling.md:12]:
```suggestion
To check the feature-gates for the Cluster API controller run the following command:
```

File: docs/book/src/topics/rosa/enabling.md [hunk 6]
```diff
@@ -4,5 +4,33 @@ To enable support for ROSA clusters, the ROSA feature flag must be set to true.
...
+check the feature gate container args it should has the MachinePool=true as below.
+
+```yaml
+spec:
+      containers:
+      - args:
+        - --feature-gates=MachinePool=true,ClusterResourceSet=true,ClusterTopology=true,RuntimeSDK=false,MachineSetPreflightChecks=false
+```
+
+To check the feature-gates for the capa contoller Run the following command.
```
Comment (vincepri) [inline docs/book/src/topics/rosa/enabling.md:25]:
```suggestion
To check the feature gates for the Cluster API AWS controller run the following command:
```

**Comment by vincepri** [comment]
/retitle 📖 Improve ROSA quickstart with MachinePool and troubleshooting guide

File: docs/book/src/topics/rosa/enabling.md (hunk 1 above)
Comment (serngawy) [reply docs/book/src/topics/rosa/enabling.md]: done

File: docs/book/src/topics/rosa/enabling.md (hunk 4 above)
Comment (serngawy) [reply docs/book/src/topics/rosa/enabling.md]: done

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**